ThroughputTest/README.md

```diff
@@ -0,0 +1,130 @@
+the street can handle a maximum capacity of cars at the same time, but at the cost of individual speed. 
+The goal of this sample is to find out where the capacity limits are. 
+
+As discussed in the [product documentation](https://docs.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement), 
+network latency has very significant impact on the achievable throughput. If you are running this sample from your
+development workstation, throughput will be substantially lower than from within an Azure VM. If you want to 
+test limits for a scenario where the Service Bus client will reside inside Azure, you should also run this test 
+inside Azure on a Windows or Linux VM that resiodes within the same region as the Service Bus namespace you 
+want to test.
+
+In an in-region setup, and with ideal parameters, you can achieve send rates exceeding 20000 msg/sec at 1024 bytes per message
+from a single client, meaning that a 1GB queue will fill up in under a minute. Also be aware that receive operations are
+generally more costly and therefore slower than send operations, which means that a test with maximum send 
+pressure (several senders using batching) may not be sustainable for longer periods because the receivers might not be 
+able to keep up.
+
+## Building the Tool
+
+The tool is a .NET Core project that can produce standalone executables for Linux and Windows, whereby we'll assume x64 targets. 
+
+For Linux, with the .NET Core 2.0 SDK installed, run: 
+
+```
+	dotnet publish -c Release -f netcoreapp2.0 -r linux-x64
+```
+The output application can be found in the ```bin/Release/netcoreapp2.0/linux-x64/publish```subdirectory.
+
+For Windows, run:
+
+```
+	dotnet publish -c Release -f netcoreapp2.0 -r win-x64
+```
+The output application can be found in the ```bin\Release\netcoreapp2.0\win-x64\publish```subdirectory.
+
+## Running the Tool
+
+You can run the tool locally from your own machine or you can run it from within an Azure VM. You should run the 
+tool on the platform you are targeting, because performance differs between Linux and Windows due to their different
+I/O architectures and different implementation of the platform layer in .NET Core.  
+
+The only required command line arguments are a connection string and a send path. If the connection string includes 
+an EntityPath property for a queue, the send path can be omitted. If the receive path is not given, the tool assumes
+the send path and the receive path to be the same, which is sufficient for queues. For topics, the subscription(s) must 
+be given with explicit receive path options.
+
+You can test any valid Service Bus topology with this tool, including Topics with subscriptions and chained 
+entities with auto-forward set up between them. Therefore, the send path and the receive path do not have to be 
+on the same entity. You can also receive from dead-letter queues. 
```

Comment:
>from dead-letter queues.  [](start = 41, length = 25)

Maybe clarify that you can't test accross multiple topics easily meaning you can have send to topic 1, receive from topic 2 subscription 1 (even if it doesn't make sense) but not have e.g. 3 queues you send to and receive too or did I miss that?